### PR TITLE
Editor store + make Zen smarter about errors (stricter error checking, show more errors, human readable)

### PR DESCRIPTION
**Pull request by DivergentEuropeans**
- prints more errors (zwe init + zwe install + jcl job submission)

TODO: JCL job submission error may be long. This error should go into an Editor too. We just don't show the Editor button in the JCL submission step, so I'll think about it more 

- change Zen to use `&&` instead of `;` so multiple commands don't fail at once, causing potential harm
- makes it so Zen doesn't let you proceed normally if zwe install, zwe init apfauth, or zwe init security fails

we more correctly check for if an error occurred 
(via text search-zwe z/OS Unix failure trigger instead of undependable status/rc that is sometimes `0` when it shouldn't be `0`)

- we more correctly show error to the user, via opening the filtered, relevant portion, via the Editor
- Because we use the Editor, to store outputs sent to/received from the Editor and to not be lost after session refresh/close, we made an Editor store
- ^the above also caused me to do some refactoring of the Stores (aka remove code duplication, put common code in a `DefaultStore`, all `Stores `build from the `DefaultStore`)
- minor typo/method renames, for example: 
`ConnectionStore.set` --> `ConnectionStore.setAndValidate` because it goes through schema validation (unchanged behaviour by me, just making things make more fucking sense)

Here's an example of these new smarts together, via a failed Apfauth step:

image

Failed job submission example:

image

**Note: the Editor stuff is not ALL hooked up to the Editor Store. This requires more code changes & can be in a future PR. The only actions right now hooked up to the store are the zwe command/job output.**

### CODE REVIEW

File: src/renderer/components/stages/InitApfAuth.tsx [hunk 1]
```diff
@@ -107,22 +112,39 @@ const InitApfAuth = () => {
...
-        dispatch(setInitializationStatus(false));
-        dispatch(setApfAuthStatus(false));
+        apfAuthProceedActions(false);
+        toggleProgress(false);
+        // TODO: Test this
+        //alertEmitter.emit('showAlert', err.toString(), 'error');
         stages[STAGE_ID].subStages[SUB_STAGE_ID].isSkipped = true;
         stages[STAGE_ID].isSkipped = true;
-        console.warn('zwe init apfauth failed');
+        console.warn('zwe init apfauth failed', err);
```
Comment (DivergentEuropeans) [inline src/renderer/components/stages/InitApfAuth.tsx]: Add a stringify here if err isn't a string

File: src/renderer/components/stages/Planning.tsx [hunk 2]
```diff
@@ -748,7 +750,7 @@ Please customize the job statement below to match your system requirements.
                       id="zosmf-host"
                       required
                       style={{marginLeft: 0}}
-                      label="z/OSMF Host"
+                      label="z/OSMF host"
```
Comment (DivergentEuropeans) [inline src/renderer/components/stages/Planning.tsx]: change this back to capital case

File: src/renderer/components/stages/Security.tsx [hunk 3]
```diff
@@ -109,22 +108,39 @@ const Security = () => {
...
+        toggleProgress(false);
         clearInterval(timer);
-        dispatch(setNextStepEnabled(false));
-        dispatch(setSecurityStatus(false));
-        dispatch(setInitializationStatus(false));
+        securityProceedActions(false);
         stages[STAGE_ID].subStages[SUB_STAGE_ID].isSkipped = true;
         stages[STAGE_ID].isSkipped = true;
-        console.warn('zwe init security failed');
+        console.warn('zwe init security failed', err);
```
Comment (DivergentEuropeans) [inline src/renderer/components/stages/Security.tsx]: stringify point here

File: src/renderer/components/stages/installation/Installation.tsx [hunk 4]
```diff
@@ -122,44 +118,53 @@ const Installation = () => {
...
-        }).catch(() => {
-          clearInterval(timer);
-          dispatch(setNextStepEnabled(false));
-          dispatch(setInitializationStatus(false));
-          dispatch(setDatasetInstallationStatus(false));
+          installProceedActions(false);
           stages[stageId].subStages[subStageId].isSkipped = true;
           stages[stageId].isSkipped = true;
-          console.warn('Installation failed');
+          console.warn('Installation failed', err);
```
Comment (DivergentEuropeans) [inline src/renderer/components/stages/installation/Installation.tsx]: stringify?

**Review by timgerstel — APPROVED** [review]

**Review by DivergentEuropeans — COMMENTED** [review]
From my code review w/ @timgerstel

**Pull request merged**